### PR TITLE
optimize: compatible with lower versions of SPI

### LINKED ISSUE #6307: Compatible with lower versions of SPI
## Why you need it?
 Is your feature request related to a problem? Please describe in details

When the user upgrades from io.seata to org.apache.seata and the user implements a lower version of the io.seata.xxx interface, the user's SPI implementation needs to be compatible without the user having to change any code.

## How it could be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.

### PR DESCRIPTION
**Pull request by xingfudeshi**
- [x] I have registered the PR [changes](../changes).

### Ⅱ. Does this pull request fix one issue?

fixes #6307

### CODE REVIEW

File: compatible/pom.xml [hunk 1]
```diff
@@ -20,33 +20,52 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-    <groupId>io.seata</groupId>
-    <version>2.1.0-SNAPSHOT</version>
+    <parent>
+        <groupId>org.apache.seata</groupId>
+        <artifactId>seata-parent</artifactId>
+        <version>${revision}</version>
```
Comment (slievrly) [inline compatible/pom.xml]: This place can't be changed to any other package name.The original used by data is required to be retained.

File: common/src/main/java/org/apache/seata/common/loader/EnhancedServiceLoader.java [hunk 2]
```diff
@@ -43,13 +43,25 @@
...
 
     /**
      * Class->InnerEnhancedServiceLoader map
      */
     private static final ConcurrentMap<Class<?>, InnerEnhancedServiceLoader<?>> SERVICE_LOADERS =
             new ConcurrentHashMap<>();
 
+    private static <S> Class<S> getCompatibleService(Class<S> originService) {
+        String apacheSeataName = originService.getName();
+        String ioSeataName = apacheSeataName.replace(APACHE_SEATA_PACKAGE_NAME, IO_SEATA_PACKAGE_NAME);
```
Comment (slievrly) [inline common/src/main/java/org/apache/seata/common/loader/EnhancedServiceLoader.java:58]: Explicitly identify the Apache Seata SPI using sartwith or other methods.

Comment (slievrly) [reply common/src/main/java/org/apache/seata/common/loader/EnhancedServiceLoader.java:58]: If both exist, what is the priority?

File: compatible/src/main/java/io/seata/core/compressor/Compressor.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.seata.core.compressor;
+
+/**
+ * The type Compressor
+ *
+ */
+@Deprecated
```
Comment (slievrly) [inline compatible/src/main/java/io/seata/core/compressor/Compressor.java]: Compatible modules do not need @Deprecated, all are deprecated.

Comment (xingfudeshi) [reply compatible/src/main/java/io/seata/core/compressor/Compressor.java]: Ok

File: compatible/pom.xml (hunk 1 above)
Comment (xingfudeshi) [reply compatible/pom.xml]: Ok

File: common/src/main/java/org/apache/seata/common/loader/EnhancedServiceLoader.java (hunk 2 above)
Comment (xingfudeshi) [reply common/src/main/java/org/apache/seata/common/loader/EnhancedServiceLoader.java:58]: If both package names exist, the current one is based on io. seata, and the package for org. apache. seata will only be loaded if it cannot be loaded. In addition, I believe that there should not be two types of package names at the same time, as compatibility itself is to provide users with a transitional solution, not a persistent compatibility solution. If users need to customize a new SPI implementation, they should switch to the package name of Apache seata as soon as possible.

**Review by slievrly — COMMENTED** [review]
I suggest changing the SPI loading priority here.
1. Load the Apache Seata SPI first. Because Apache Seata is the final state of the future.
2. If the Apache seata SPI cannot be found, output a warn log and try to load the io.seata SPI.
3. If the io.seata SPI also fails to be loaded, handle the exception according to the logic in the original SPI.

**Comment by xingfudeshi** [comment]
> I suggest changing the SPI loading priority here.
> 
> 1. Load the Apache Seata SPI first. Because Apache Seata is the final state of the future.
> 2. If the Apache seata SPI cannot be found, output a warn log and try to load the io.seata SPI.
> 3. If the io.seata SPI also fails to be loaded, handle the exception according to the logic in the original SPI.

Ok

**Comment by xingfudeshi** [comment]
Done. @slievrly

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**